minigalaxy/installer.py

```diff
@@ -16,7 +16,14 @@ def get_available_disk_space(location):
     symlinks to disks with sufficient space are correctly measured. Note this is
     a linux-specific command."""
     absolute_location = os.path.realpath(location)
-    disk_status = os.statvfs(os.path.dirname(absolute_location))
+    while True:
```

Comment:
I would write this in a slightly different way to make sure we're not going up all the way to `/`, which could be a different filesystem.

Something like this should work:
```
if os.path.isdir(absolute_location):
    disk_status = os.statvfs(absolute_location)
else:
    disk_status = os.statvfs(os.path.dirname(absolute_location))
```

I might be missing something, though. Let me know if that's the case, then I'll merge this as is.

Reply:
> I would write this in a slightly different way to make sure we're not going up all the way to `/`, which could be a different filesystem.

I don't understand the concern. We go only up to `/` if no part of the path actually exists. Any part of the path could be on a different filesystem. But we never go there, because we stop at the first existing node.

> I might be missing something, though. Let me know if that's the case, then I'll merge this as is.

I wasn't sure if it's always guaranteed that the first parent directory exists. I guess the current code makes the assumptions and it wouldn't cause any new problems.

Reply:
Actually, I did a quick test just now and it does not really matter if os.statvfs is used on a directory or a file, the output should be the same.

So instead of this change, we can just change the original line 19 to not use dirname. That should solve the issue.

Could you do that and confirm that this solves the issue for you?

Reply:
To explain my thinking here. We need statvfs information from where we want to write. Right now we get information from one directory higher than that, which can be a different file system (we know for sure it is on Flatpak). I'm worried going up further will just give us wrong information, like right now.

Thanks a lot for investigating this issue and putting work into it :)

Reply:
I have simplified the PR and just always call it in the parent directory. I hope this is more in line with your expectations.

> Actually, I did a quick test just now and it does not really matter if os.statvfs is used on a directory or a file, the output should be the same.

Yes, but it is never called with a file. It is called with either `config.install_dir` or `installer.install_dir`. In the latter case the directory does not exist yet. 

> I'm worried going up further will just give us wrong information, like right now.

But we do not go further. If it is another file system, the path is guaranteed to exist up to this point, otherwise there can be no mount point for another file system.